lib/services/user-service-base.ts

```diff
@@ -0,0 +1,84 @@
+export abstract class UserServiceBase {
+	private userData: IUserData;
+	protected userFilePath: string;
+
+	private get $nsCloudServerAccountsService(): IServerAccountsService {
+		return this.$injector.resolve<IServerAccountsService>("nsCloudServerAccountsService");
```

Comment:
IMO you can put all of the dependencies inside the class' `constructor` instead of having them all as properties.
Property dependencies exist primarily because of cyclic chains